cli/azd/cmd/package.go

```diff
@@ -128,10 +132,20 @@ func (pa *packageAction) Run(ctx context.Context) (*actions.ActionResult, error)
 		return nil, err
 	}
 
+	var stepMessage string
+	progressFilter := func(msg messaging.Message) bool {
+		return msg.Type == project.ProgressMessage
+	}
+	progressSubscription := pa.subscriber.Subscribe(ctx, progressFilter, func(msg messaging.Message) {
+		progressMessage := fmt.Sprintf("%s (%s)", stepMessage, msg.Value)
+		pa.console.ShowSpinner(ctx, progressMessage, input.Step)
+	})
+	defer progressSubscription.Close()
```

Comment:
Simply subscribe to the types of messages you care about and automatically receive a callback when a new message matching your filter becomes available.